core/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/pipeline/PipelineServiceImpl.java

```diff
@@ -106,17 +115,20 @@ public CompletableFuture<StatusResult<Void>> transfer(DataSink sink, DataFlowReq
             return noSourceFactory(request);
         }
         var source = sourceFactory.createSource(request);
+        monitor.debug(() -> format("transferring from %s to %s.", source, sink));
         return sink.transfer(source);
     }
 
     @Override
     public void registerFactory(DataSourceFactory factory) {
         sourceFactories.add(factory);
+        monitor.info(() -> format("%s was registered: %s.", DataSourceFactory.class.getSimpleName(), factory.getClass().getSimpleName()));
```

Comment:
Please remove this statement as it is extraneous - the runtime will already emit a message that the extension was loaded, and it doesn't need to record a factory was registered.

Reply:
I still want debug logging for extensions dynamically switching  and registering source/sink based on the condition.

Reply:
We explicitly designed extensions not to be dynamic like OSGI. The `PipelineService` is also designed not to be dynamic.

An event for the loaded extension will already be emitted, so doing so for the factory offers no value. Please remove it.

Reply:
removed.